File: reprexes/test_read_write_geopandas.py
# Aim: benchmark geopandas reading of gpkg and geojson files

file_path1 = "test.geojson"
file_path2 = "test.gpkg"

import geopandas as gpd

gdf1 = gpd.read_file(file_path1)
gdf2 = gpd.read_file(file_path2)

# timings and output to csv
import timeit
import csv

def read_gpkg():
    gpd.read_file(file_path2)

def read_geojson():
    gpd.read_file(file_path1)

def main():
    with open('benchmark.csv', 'w', newline='') as csvfile:
        writer = csv.writer(csvfile, delimiter=',')
        writer.writerow(['file', 'time'])
        writer.writerow(['test.geojson', timeit.timeit(read_geojson, number=10)])
        writer.writerow(['test.gpkg', timeit.timeit(read_gpkg, number=10)])

main()

File: reprexes/test_read_write_geopandas.R
# Aim: test read/write speed with geopandas

library(reticulate)

# Import the geopandas package:
geopandas = import("geopandas")

# Save a geojson and then a gpkg file:
test_object = stplanr::routes_fast_sf

# Save as geojson:
sf::write_sf(test_object, "test.geojson", delete_dsn = TRUE)
sf::write_sf(test_object, "test.gpkg", delete_dsn = TRUE)

# Test reading with geopandas:
runtime = system.time({
  output = geopandas$read_file("test.geojson")
})

benchmark = bench::mark(
  check = FALSE,
  geopandas_geojson = geopandas$read_file("test.geojson"),
  geopandas_gpkg = geopandas$read_file("test.gpkg"),
  sf_geojson = sf::read_sf("test.geojson"),
  sf_gpkg = sf::read_sf("test.gpkg")
  )

ggplot2::autoplot(benchmark)
